src/test/java/org/skyscreamer/jsonassert/JSONAssertTest.java

```diff
@@ -640,6 +641,94 @@ public void testAssertNotEqualsString2JsonComparator() throws IllegalArgumentExc
                 new RegularExpressionValueMatcher<Object>("\\d"))
         ));
     }
+
+    @Test
+    public void testAssertEquals2JSONArraysCompareIgnoringCustomizationsFields() 
+    		throws IllegalArgumentException, JSONException {
+    	// Test for issue 109 (https://github.com/skyscreamer/JSONassert/issues/109)
+        String json1 = "{\n" +
+                "  	\"root\": {\n" +
+                "  	\"level\": {\n" +
+                "  		\"array\": [{\n" +
+	            "  			\"field1ToIgnore\": \"ignore_value1\",\n" +
+	            "  			\"field2\": \"valueA2\",\n" +
+                "  			\"field3\": \"valueA3\"\n" +
+                "  			}, {\n" +
+                "  			\"field1ToIgnore\": \"ignore_value2\",\n" +
+                "  			\"field2\": \"valueB2\",\n" +
+                "  			\"field3\": \"valueB3\"\n" +
+                "  		}]\n" +
+                "  	}\n" +
+                "  	}\n" +
+                "  }\n";
+        String json2 = "{\n" +
+                "  	\"root\": {\n" +
+                "  	\"level\": {\n" +
+                "  		\"array\": [{\n" +
+                "  			\"field1ToIgnore\": \"ignore_value3\",\n" +
+                "  			\"field2\": \"valueB2\",\n" +
+                "  			\"field3\": \"valueB3\"\n" +
+                "  			}, {\n" +
+                "  			\"field1ToIgnore\": \"ignore_value4\",\n" +
+                "  			\"field2\": \"valueA2\",\n" +
+                "  			\"field3\": \"valueA3\"\n" +
+                "  		}]\n" +
+                "  	}\n" +
+                "  	}\n" +
+                "  }\n";
+        
+        JSONAssert.assertEquals("Message", json1, json2, 
+            new CustomComparator(
+                JSONCompareMode.LENIENT, 
```

Comment:
Looking at this whole change critically, I think the major issue is that _lenient_ and _customizations_ kind of don't agree with each other. When you're being lenient, you can allow for extra elements in lists, so why is an element that differs via fields that are customised treated as a match that's in the right place, rather than a mismatch that's _additional_?